StreamChatClient_v3/WebSocketClient/Engine/WebSocketEngine_tests.swift

```diff
@@ -0,0 +1,65 @@
+//
+// WebSocketEngine_tests.swift
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+@testable import StreamChatClient_v3
+
+class WebSocketEngineMock: WebSocketEngine {
+  var request: URLRequest
+  var isConnected: Bool = false
+  var callbackQueue: DispatchQueue
+  weak var delegate: WebSocketEngineDelegate?
+
+  /// How many times was `connect()` called
+  var connect_calledCount = 0
+
+  /// How many times was `disconnect()` called
+  var disconnect_calledCount = 0
+
+  /// How many times was `sendPing()` called
+  var sendPing_calledCount = 0
+
+  convenience init() {
+    self.init(request: .init(url: URL(string: "test_url")!), callbackQueue: .main)
+  }
+
+  required init(request: URLRequest, callbackQueue: DispatchQueue) {
+    self.request = request
+    self.callbackQueue = callbackQueue
+  }
+
+  func connect() {
+    connect_calledCount += 1
+  }
+
+  func disconnect() {
+    disconnect_calledCount += 1
+  }
```

Comment:
Do you need to change `isConnected` here?

Reply:
This is just a mock. In the situation when you need to change the `isConnected` value, you can do it directly by assigning the value from tests:
```
engineMock.isConnected = false
```
Ideally, all mocks are super dumb and just provide the API needed.